### DISCUSSION TITLE
How does `remult` injection into BackendMethods in Deno work?

### DISCUSSION THREAD

**Discussion by ChrisRimmer** [noam-honig/radweb#181] [Q&A] [answered]
Is there anything specific I have to do to prompt Remult to inject a `remult` when calling BackendMethods in Deno? I'm running the below code as described in the docs here https://remult.dev/docs/backendMethods.html#deno-and-other-run-times-that-do-not-support-asynclocalstorage but there is no third argument being passed in, when I check in my debugger I very definitely only get two arguments passed at call-time.

Filing this as a discussion for now as I'm not convinced it's a bug, but I'll bring it over to issues if necessary.

```typescript
import { Model } from "model.ts"

export class ThingController {
    @BackendMethod({allowed: true})
    static async doThing(arg1: string, arg2: string, remult?: Remult) {
        const repo = remult!.repo(Model)

**Comment by noam-honig** [comment, marked as answer]
Hi @ChrisRimmer,

There are several options you can use:
# Option 1 - Add "emitDecoratorMetadata" to deno:
Change `deno.json`
```json
{
  "tasks": {
    "start": "deno run -A --watch=static/,routes/ dev.ts"
  },
  "compilerOptions": {
    "experimentalDecorators": true,
    "emitDecoratorMetadata":true
    
  },
  "importMap": "./import_map.json"
}

```
It spits out a warning saying that the option was ignored - but it is not ignored, and it works.

# option 2 - specify parameter type:
In the `BackendMethod` decorator, specify the parameter types:
```ts
 @BackendMethod({allowed: true, paramTypes: [String,String,Remult})
```

Check it out and let me know

**Reply by ChrisRimmer** [reply]
Option 1 worked like a charm, thanks!
